mr/mr_main.h

```diff
@@ -39,10 +39,10 @@ class PipelineMain {
   std::unique_ptr<MainInitGuard> guard_;  // Must be first to be destructed last.
 
   std::unique_ptr<util::IoContextPool> pool_;
-  std::unique_ptr<Pipeline> pipeline_;
+  std::vector<std::unique_ptr<Pipeline>> pipelines_;
```

Comment:
I do not understand why you need to store multiple pipelines & runners inside PipelineMain.
Maybe just introduce `void ResetPipeline()` that resets (reallocates) the pipeline object? 
it won't work?